scRNA-seq-advanced/setup/integration-setup.Rmd

```diff
@@ -0,0 +1,566 @@
+
+# Let's have a peek at the rowData:
+# `mean` and `detected` are SCE-specific, but `gene_symbol` is not! Bookkeeping!
+purrr::map(sce_list, 
+           ~names(rowData(.)))
+```
+
+We want to format each SCE to meet our criteria:
+```{r}
+format_sce_list <- function(sce, sce_name, shared_genes) {
+  
+  # Reduce to shared genes
+  sce <- sce[shared_genes,]
+
+  # Update the colData row names so we can know where cells came from back
+  # Alternatively we can just get rid of a lot of this?
+  # glue is great!
+  rownames(colData(sce)) <- glue::glue("{sce_name}-{rownames(colData(sce))}")
+  
+  # Update the rowData names, except `gene_symbol`
+  names(rowData(sce)) <- ifelse(names(rowData(sce)) == "gene_symbol",
+                                "gene_symbol",
+                                glue::glue("{sce_name}-{rownames(rowData(sce))}")
+    
+  ) 
+  # Add in sample-level information as stand-alone column 
+  colData(sce)$sample <- sce_name
+  
+  # Return 
+  return(sce)
+}
+
+
+# Prepare the sce_list for combining
+sce_list <- purrr::imap(sce_list, 
+                        format_sce_list,
+                        shared_genes)
+
+# Combine!
+combined_sce <- do.call(cbind, sce_list)
+combined_sce
+```
+
+You'll see that there are `PCA` and `UMAP` dimension reductions in this combined SCE.
+These represent the original reductions that were performed on _each individual library_.
+We can also calculate PCA and UMAP for the combined whole, which we will save with different names `combined_PCA` and `combined_UMAP` to distinguish.
+We can compare these UMAP embeddings to those calculated after performing integration.
+
+```{r}
+# calculate high-variance genes to use for dimension reduction
+gene_variance <- scran::modelGeneVar(combined_sce, 
+                                     # For combined SCEs, specify the group column groups:
+                                     block = combined_sce$sample) 
+hvg_list <- scran::getTopHVGs(gene_variance,
+                              n = 2000) 
+  
+# Add PCA and UMAP into the SCE
+combined_sce <- scater::runPCA(combined_sce, 
+                               subset_row = hvg_list, 
+                               # Make sure not to overwrite the existing PCA!
+                               name = "combined_PCA") %>%
+  # again, make sure to use the right names
+  scater::runUMAP(dimred = "combined_PCA", 
+                  name = "combined_UMAP")
+
+combined_sce
+```
+
+
+
+## Pre-integration visualization
+
+To get a sense of our data _before_ integrating, let's look at UMAPs for each individual library and the combined SCE.
+
+
+```{r}
+plot_library_umap <- function(sce, sce_name) {
+  scater::plotReducedDim(sce,
+                         dimred = "UMAP", 
+                         point_size = 0.5, 
+                         point_alpha = 0.5) +
+    ggplot2::ggtitle(sce_name)
+}
+
+umap_list <- purrr::imap(
+  sce_list,
+  plot_library_umap
+)
+
+cowplot::plot_grid(plotlist = umap_list, nrow = 2)
```

Comment:
>  because I see you did plot these colors in the next plot!

Reply:
> because I see you did plot these colors in the next plot!

These are just the default colors the function applies, and `aes` is the same so it works out that way. "facet after the fact" to me sounds like a cowplot/patchwork thing though since we're dealing with multiple ggplot2 objects here. Am I understanding correctly?

Reply:
Sadly nope, no can `facet`. We'd have to plot it from a data frame to start, but that wrangling isn't too bad.